blog(embertimes154): ember-stargate (#697)
ember-learn/ember-blog: source/2020-06-26-the-ember-times-issue-154.md

<SAYING-HELLO-IN-YOUR-FAVORITE-LANGUAGE> Emberistas! 🐹

<SOME-INTRO-HERE-TO-KEEP-THEM-SUBSCRIBERS-READING>
Check out the new addon ember-stargate 🌟!
READMORE

---

## [Get Ready for ember-stargate 🌟](https://twitter.com/simonihmig/status/1274066847873273859)

[Simon Ihmig (@simonihmig)](https://github.com/simonihmig) announced a new addon called [ember-stargate](https://github.com/kaliber5/ember-stargate) – a modern and lightweight take on portals in Ember.

“Portals” are a way to render things in a different place of the DOM tree than they are logically defined in the app. There are a number of existing solutions in the Ember ecosystem for the same problem such as [ember-wormhole](https://github.com/yapplabs/ember-wormhole), [ember-elsewhere](https://github.com/ef4/ember-elsewhere), the native `{{in-element}}` (previously using the [ember-in-element-polyfill](https://github.com/ember-polyfills/ember-in-element-polyfill)).

> The `ember-stargate` addon aims to solve this problem without using private APIs, Ember's `component` helper, or having to jump through hoops to implement. All you need to make a portal work is to define the content that should be sent through the portal using `<Portal @target="some-target">`, and the target - identified by its name - where it should be rendered to using `<PortalTarget @name="some-target">`. For example, you have 2 templates the first being the route template and the second being the `application.hbs`:

Reviewer: ```suggestion
The `ember-stargate` addon aims to solve this problem without using private APIs, Ember's `component` helper, or having to jump through hoops to implement. All you need to make a portal work is to define the content that should be sent through the portal using `<Portal @target="some-target">`, and the target - identified by its name - where it should be rendered to using `<PortalTarget @name="some-target">`.

For example, suppose you want to update the footer in `application.hbs` from a child route:
```

> Route:

Reviewer: ```suggestion
Route:

```

```hbs
<Portal @target="footer-title">Step 1</Portal>
```

> Application:

Reviewer: ```suggestion
Application:

```
